docs/advanced.md

```diff
@@ -471,3 +471,50 @@ If you do need to make HTTPS connections to a local server, for example to test
 >>> r
 Response <200 OK>
 ```
+
+## Retries
+
+Communicating with a peer over a network is by essence subject to errors. HTTPX provides built-in retry functionality to increase the resilience to connection issues.
+
+### Enabling retries
+
+Retries are disabled by default. You can enable them on a client instance using the `retries` parameter:
+
+```python
+# Retry at most 3 times on connection failures.
+client = httpx.Client(retries=3)
+```
+
+When retries are enabled, HTTPX will retry sending the request up to the specified number of times. This behavior is restricted to **connection failures only**, i.e.:
+
+* Failures to establish or acquire a connection (`ConnectTimeout`, `PoolTimeout`).
+* Failures to keep the connection open (`NetworkError`).
+
+!!! important
+    HTTPX will **NOT** retry on failures that aren't related to establishing or maintaining connections. This includes in particular:
+
+    * Errors related to data transfer, such as `ReadTimeout` or `ProtocolError`.
+    * HTTP error responses (4xx, 5xx), such as `429 Too Many Requests` or `503 Service Unavailable`.
```

Comment:
Hmm. I feel like retrying on status codes is still valuable for a retry config.

Could you add a doc on how to extend this to handle that, if it won't be available by default? (Those codes in particular I think are the primary use cases in fact)

Reply:
> Could you add a doc on how to extend this to handle that, if it won't be available by default?

Well so, the idea behind this PR is to not even provide the ability to extend *just yet*. I agree this can fall in scope for us though, as well as handling `Retry-After`…

Reply:
I'd rather we consider adding support for those cases and `Retry-After` as a follow-up task, in any case. :-) Let's get something really simple but API-proof in, first?

Reply:
Yeah, agreed. I personally am looking for Retry support for those cases (well, server errors at least), so I'll be hoping that gets added as a follow-up. Definitely want something fairly bullet-proof to build on though!